src/python/pants/backend/python/target_types.py

```diff
@@ -725,6 +725,191 @@ class PexVenvHermeticScripts(BoolField):
+        `platform-file-suffix` forces the scie target platform name to be
+        added as a suffix of the output filename; Specifying
+        `platform-parent-dir` places the scie in a sub- directory with the
+        name of the platform it targets."""
+    )
+
+
+class PexScieBusyBox(StringField):
+    alias = "scie_busybox"
+    default = None
+    help = help_text(
+        """
+        Make the PEX scie a BusyBox over the specified entry points. The entry
+        points can either be console scripts or entry point specifiers. To
+        select all console scripts in all distributions contained in the PEX,
+        use `@`. To just pick all the console scripts from a particular
+        project name's distributions in the PEX, use `@<project name>`; e.g.:
+        `@ansible-core`. To exclude all the console scripts from a project,
+        prefix with a `!`; e.g.: `@,!@ansible-core` selects all console
+        scripts except those provided by the `ansible- core` project. To
+        select an individual console script, just use its name or prefix the
+        name with `!` to exclude that individual console script. To specify an
+        arbitrary entry point in a module contained within one of the
+        distributions in the PEX, use a string of the form
+        `<name>=<module>(:<function>)`; e.g.: 'run- baz=foo.bar:baz' to
+        execute the `baz` function in the `foo.bar` module as the entry point
+        named `run-baz`.
+
+        A BusyBox scie has no default entrypoint; instead, when run, it
+        inspects argv0; if that matches one of its embedded entry points, it
+        runs that entry point; if not, it lists all available entrypoints for
+        you to pick from. To run a given entry point, you specify it as the
+        first argument and all other arguments after that are forwarded to
+        that entry point. BusyBox PEX scies allow you to install all their
+        contained entry points into a given directory.  For more information,
+        run `SCIE=help <your PEX scie>` and review the `install` command help.
+
+        NOTE: This is only available for formal Python entry points
+        <https://packaging.python.org/en/latest/specifications/entry-points/>
+        and not the informal use by the `pex_binary` field `entry_point` to
+        run first party files.
+        """
+    )
+
+
+class PexScieBusyboxPexEntrypointEnvPassthrough(TriBoolField):
+    alias = "scie_busybox_pex_entrypoint_env_passthrough"
+    required = False
+    default = None
+    help = help_text(
+        """ When creating a busybox, allow overriding the primary entrypoint
+        at runtime via PEX_INTERPRETER, PEX_SCRIPT and PEX_MODULE. Note that
+        when using the `venv` execution mode this adds modest startup overhead
+        on the order of 10ms.  """
+    )
+
+
+class PexSciePlatformField(StringSequenceField):
+    alias = "scie_platform"
+    valid_choices = (
```

Comment:
I did `valid_choices` here since I think it will help people get the format right (`aarch64` instead of the `arm64` used by other parts of Pants for example) and new arches don't come along *that* often.